processor/memorylimiterprocessor/memorylimiter_test.go

```diff
@@ -31,334 +29,70 @@ import (
 // TestMetricsMemoryPressureResponse manipulates results from querying memory and
 // check expected side effects.
 func TestMetricsMemoryPressureResponse(t *testing.T) {
```

Comment:
Does this test have to be rewritten for the split? If not, let's keep the original version. I understand the subtests are better, but changing them in this PR makes the review harder and may hide some issues. Same goes to `TestTracesMemoryPressureResponse` and `TestLogsMemoryPressureResponse` tests

Reply:
yes because the modifying of values required are no longer being exposed. They got moved to `internal/memorylimiter`. The original test are also moved there, this is just an insurance test.

Reply:
What values? I see monkey patching of `GetMemoryFn` which you do in the new version of the tests as well

Reply:
in only patch `GetMemoryFn`, but would also need to expose `readMemStatsFn` for the previous tests to work. since the test is copied to `internal/memorylimiter_test`, is it still needed to in the processor?

Reply:
Thanks for clarifying. Makes sense now.